tests/test_device.py

```diff
@@ -5,6 +5,7 @@
 import warnings
 import numpy as np
 import pytest
+plt = pytest.importorskip("matplotlib.pyplot")
```

Comment:
Will this skip all the tests in this file?

Reply:
You are right this skips the whole test module. Importation has been moved inside the test function.